idaes/models/properties/modular_properties/base/generic_property.py

```diff
@@ -1340,81 +1363,118 @@ def initialize(
+                        for j in blk[k].component_list))
+    
+                p_crit = value(sum(blk[k].mole_frac_comp[j] *
+                                   blk[k].params.get_component(j).pressure_crit
+                                   for j in blk[k].component_list))
+                v_crit = value(
+                    sum(blk[k].mole_frac_comp[j] *
+                        p_config.components[j]['parameter_data']['volume_crit'][0]
+                        for j in blk[k].component_list))
+                Z_crit = value(
+                    sum(blk[k].mole_frac_comp[j] *
+                        p_config.components[j]['parameter_data']['compress_factor_crit'][0]
+                        for j in blk[k].component_list))
+    
+                blk[k].pressure_crit_mix.value = value(
+                    Z_crit * const.gas_constant * 
+                    blk[k].temperature_crit_mix / v_crit)
+    
+                # for ASU system:
+                # blk[k].temperature_crit_mix.value = 136.560995169
+                # blk[k].pressure_crit_mix.value = 3953826.947276
+                
+                for c in blk[k].component_objects(Constraint):
+                    if c.local_name in ('A_crit', 'B_crit'):
+                        c.activate()
+                    else:
+                        c.deactivate()
+        
+        if p_config.supercritical:
+            with idaeslog.solver_log(solve_log, idaeslog.DEBUG) as slc:
+                res = solve_indexed_blocks(opt, [blk], tee=slc.tee)
+            init_log.info(
+                "Critical point initialization complete: {}."
+                .format(idaeslog.condition(res))
+            )
+
+        # ---------------------------------------------------------------------
+        # Initialize P+, P- and pbar
+        for k in blk.keys():
+            if blk[k].params.config.supercritical:
+                if (blk[k].params.config.phases_in_equilibrium is not None and
+                        (not blk[k].config.defined_state or blk[k].always_flash)):
+                    for pp in blk[k].params._pe_pairs:
+                        blk[k].params.config.phase_equilibrium_state[pp] \
+                            .calculate_pbar(blk[k], pp)
+                        print(f"pressure = {blk[k].pressure.value}\n"
+                              f"pbar = {blk[k].pbar[pp].value}")
+        
+        if blk[k].params.config.supercritical:
+            init_log.info("Pressure slacks initialization complete.")
 
         # ---------------------------------------------------------------------
         # If present, initialize bubble and dew point calculations
         for k in blk.keys():
             T_units = blk[k].params.get_metadata().default_units["temperature"]
             # Bubble temperature initialization
-            if hasattr(blk[k], "_mole_frac_tbub"):
-                blk._init_Tbub(blk[k], T_units)
+            blk._init_Tbub(blk[k], T_units)
```

Comment:
Bubble and dew points estimates are used in the initialization scheme and remain fixed. Although we would not like the user to access them as they do not participate in the formulation itself.

Reply:
Could you clarify what you mean by "remain fixed" in this case? I think we need to be clear about what is going on here, as bubble and dew points might be something the user wants to calculate, and if they see them on the model they will want to use them.

If this is just for initialization, then I would suggest that we should probably use temporary variables for these (not attached to the model), or we should treat these are yet another property and initialize and solve them as such.

Reply:
IIRC, the only reason we kept the `temperature_bubble` and `temperature_dew` variables in the model was because other unit models (in our case the tray distillation column model) were using them in their initialization routine. @jghouse88 is this accurate? We had removed them completely from `generic_property` before realizing that the tray column model needed them, so keeping them in `generic_property` and fixing them was our (perhaps temporary) solution. If we think they are necessary for the user, I am good with keeping them as yet another property.

Reply:
I think this needs to be an binary decision; either we keep them as full properties or we remove them completely (and deal with the consequences).

Right now, I am not sure what the state is, as they appear to be in some intermediate state.

Reply:
I've previously used `temperature_bubble` and `temperature_dew` in heat exchanger constraints to keep, e.g., boiling, separate from preheating and superheating. Does the new formulation support such constraints without bubble and dew properties?

Reply:
That is what we are working out - I believe we do need to keep these supported as properties which I am not sure the current implementation odes (but I could be wrong).

Reply:
The new formulation obviates the need to calculate bubble and dew points. As of now, we have kept them in to get estimates for bubble and dew temperature ONLY at the feed composition. But if one needs bubble and dew points at some intermediate stage (say the top tray, or the condenser) of a process, then we have not supported this currently. I do think we can just add an on-demand method to build and calculate the bubble and dew point subsystem for any mixture composition. That way, the VLE formulation remains free of these calculations (this is a unique selling point of the proposed formulation) and the bubble and dew calculations can be triggered by the user as required.

Reply:
Yes, that is what we need to do (add bubble and dew points as normal build-on-demand properties).

Reply:
Let me see if I can clarify a few things here:
1. The old Smooth VLE formulation depended on `T_bubble` and `T_dew` to calculate `T_bar`. The challenges with this approach were:
   a. Always carry constraints for `T_bubble` and `T_dew` 
   b. More cumbersome initialization where significant time was spent initially trying to solve for `T_bubble` and `T_dew`. 
   c. When the system is close to `T_crit`, this approach leads to convergence failures as expected. 
2. The new approach, eliminates the need for calculating `T_bubble` and `T_dew` altogether. However, to get a reasonable initial guess for `T_bar` at the start of the initialization step, @viiibhav uses an iterative approach assuming ideal mixture to **ONLY** estimate `T_bubble` and `T_dew`. This information is then used to initialize `T_bar` as an average of `T_bubble` and `T_dew`. Then the initialization proceeds and note that the `T_bubble` and `T_dew` were calculated for some initial guess of state variables which could change once the system has been initialized. Those initial values for T_bubble and T_dew should never be accessed by the user as they do not update with changing state conditions. 
3. How did we deal with no `T_bubble` and `T_dew` with tray columns especially for units like total condenser? For the total condenser, the outlet spec was changed from saying something like at `T_bubble` to adding a constraint `sum(k_i*x_i)==1`. This worked better actually. 
4. With regard to the "need to provide" these as on-demand properties will mean we carry those bubble and dew point constraints from the old formulation which negates the advantages of the modified formulation without bubble and dew points. The worry here is even though we can add those constraints to be built on-demand, we have no clue what it would do to the convergence with those constraints enabled because we have not tested it. 

After discussions with @viiibhav, we realize that `temperature_bubble` and `temperature_dew` vars need not be Pyomo variables. We will make that change.

Reply:
My concern here is points 3 and 4 - it looks like you have addressed the issue for tray columns but there might be other models (including external users) who make use of these so I am hesitant to just drop them outright. Regarding convergence, that is ultimately only a concern for those who use it.

For points 1 and 2 - we should make these internal Python variables (not Pyomo `Vars`) which are not attached to the model and use names that indicate that, e.g. `_Tbub.

Reply:
Agree on them not being Pyomo vars. I am worried of having something in there where we have not tested its efficacy or convergence. Better to raise an exception that CEOS with smooth VLE does not contain bubble and dew point vars but recommend a few alternatives to get that value with existing formulation. Thoughts?

Reply:
The issue is that that is a deprecation; we can't just remove features like that without warning or leaving a backward compatibility. And backward compatibility means continuing to support bubble and dew point calculations.

Reply:
I think bubble/dew variables are currently fully supported with the create-on-demand interface right now. It's just that they're always called when `SmoothVLE` is called and are thus created whenever we have VLE.  Currently, the config option defaults to `LogBubbleDew`, so most property packages don't even specify a method.

Last I checked with @viiibhav , he had a function that triggers their creation without putting them in any VLE `Constraint` (but the `Constraints` which define their values still exist). The idea was that, once this PR is merged, either Andrew or I could go through and clean up the various places where bubble/dew temperatures are still being used unnecessarily.